kotlin/script/setup

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+set -e
+
+# If JAVA_HOME not set then search for jdk to set JAVA_HOME to
+# This includes the JDK integrated into Android studio
+if [[ -z "$JAVA_HOME" ]]; then
+    echo "JAVA_HOME not set"
+    android_studio_path="/Applications/Android Studio.app/Contents"
+    system_java_home=$(/usr/libexec/java_home)
+    potential_java_homes=("$JAVA_HOME" "$system_java_home" "$android_studio_path/jre/Contents/Home" "$android_studio_path/jre/jdk/Contents/Home")
```

Comment:
Are these just mac/linux paths? Anything to be done for windows?

Reply:
These are just Mac paths currently, tomorrow I'll add a few common Linux paths as I am a big Linux lover.

I've been working on a different approach for Windows that'll use .ps1 powershell scripts (as bat scripts are total trash to work with), but that's a separate PR.

Reply:
*UPDATE*: ./gradlew.bat test is working on Windows now that I've checked in the Gradle wrapper folder. 
This involves checking in a 50KB gradle-wrapper.jar. I don't like it, but it is consistent with other projects and the gradle official documentation. You can read more here: https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:adding_wrapper

As a result I'm adding to the README how to run on Windows using gradlew.bat 

I'm keeping the shell scripts in the scripts folder pretty much just for consistency with the other programming languages.